plugins/version-control-systems/git/src/main/kotlin/Git.kt

```diff
@@ -294,14 +327,27 @@ class Git internal constructor() : VersionControlSystem(GitCommand) {
 
         runCatching {
             // TODO: Migrate this to JGit once https://bugs.eclipse.org/bugs/show_bug.cgi?id=580731 is implemented.
-            workingTree.runGit("submodule", "update", "--init", "--recursive", "--depth", "$GIT_HISTORY_DEPTH")
+            val updateArgs = mutableListOf("submodule", "update", "--init", "--depth", "$historyDepth").apply {
+                if (recursive) { add("--recursive") }
+            }
+
+            workingTree.runGit(*updateArgs.toTypedArray())
 
             insteadOf.forEach {
-                workingTree.runGit("submodule", "foreach", "--recursive", "git config $it")
+                val foreachArgs = mutableListOf("submodule", "foreach").apply {
+                    if (recursive) { add("--recursive") }
+                    add("git config $it")
+                }
+
+                workingTree.runGit(*foreachArgs.toTypedArray())
             }
         }.recover {
             // As Git's dumb HTTP transport does not support shallow capabilities, also try to not limit the depth.
-            workingTree.runGit("submodule", "update", "--recursive")
+            val fallbackArgs = mutableListOf("submodule", "update").apply {
```

Comment:
Our naming of this variable is a bit inconsistent: Above you name it after the `submodule` sub-command's name, but here after the code location. I guess I'd just call all of them simply `args`, but if you want to stick to more dedicated names, this should probably be called `updateArgs`, too.